webClient/src/app/editor/code-editor/monaco/monaco.service.ts

```diff
@@ -306,6 +306,11 @@ export class MonacoService {
   
   saveFile(fileContext: ProjectContext, fileDirectory?: string): Observable<void> {
     return new Observable((obs) => {
+
+      if (fileContext.model.isDataset) {
+        //TODO validation
```

Comment:
You should expand on this. What validation should we probably be doing?

Reply:
we are doing the validations in `isContentValidForDataset `method in `webClient/src/app/shared/editor-control/editor-control.service.ts`. I will modify the comment here.